lib/js/src/transport/WebSocketClientConfig.js

```diff
@@ -64,6 +65,24 @@ class WebSocketClientConfig extends TransportConfigBase {
     getPort () {
         return this._port;
     }
+
+    /**
+     * Provides the WebSocketClientConfig instance for WebEngine Application based on URL Search Params
+     * @return {WebSocketClientConfig}
+     */
+    static webEngine () {
+        let protocol = 'ws';
+        let host = 'localhost';
+        let port = 5050;
+        if (URLSearchParams && window && window.location && window.location.search) {
+            const urlParams = new URLSearchParams(window.location.search);
+            const role = (urlParams.get('sdl-transport-role') || `${protocol}-server`);
+            protocol = role.substring(0, role.length - 7);
+            host = urlParams.get('sdl-host') || host;
+            port = urlParams.get('sdl-port') || port;
+        }
+        return new WebSocketClientConfig(`${protocol}://${host}`, port);
+    }
```

Comment:
As I mentioned in my previous review, I think this logic would be better placed in the constructor rather than as a new static method. By placing the logic in the constructor we can allow developers a more consistent development experience regardless of whether or not their app is a WebEngine app.

Within the constructor, a check could be made to determine if the querystring parameters are present, and if so, use them, otherwise fall back to the passed `host` and `port` arguments in the constructor. This approach would potentially streamline a developer's local testing efforts as well.

Reply:
Done in 3e94fbb